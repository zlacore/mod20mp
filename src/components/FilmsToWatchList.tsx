import type React from 'react';
import type Film from '../utils/interfaces/Film.interface';
import FilmCard from './FilmCard';

interface FilmsToWatchProps {
  filmsToWatch: Film[];
  removeFromStorage:
    | ((
        e: React.MouseEvent<SVGSVGElement, MouseEvent>,
        currentlyOnWatchList: boolean | null | undefined,
        currentlyOnSeenItList: boolean | null | undefined,
        title: string | null
      ) => void)
    | null;
}

const FilmsToWatchList = ({
  filmsToWatch,
  removeFromStorage,
}: FilmsToWatchProps) => {
  console.log(filmsToWatch);

  return (
    <>
      <ul>
        {filmsToWatch.map((film) => (
          <FilmCard
            currentFilm={film}
            key={film.Title}
            onWatchList={true}
            removeFromStorage={removeFromStorage}
          />
        ))}
      </ul>
    </>
  );
};

export default FilmsToWatchList;
