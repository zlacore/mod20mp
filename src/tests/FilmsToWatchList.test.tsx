import pretty from 'pretty';
import { render } from '@testing-library/react';
import FilmsToWatchList from '../components/FilmsToWatchList';
import type Film from '../utils/interfaces/Film.interface';

const films: Film[] = [{
  Title: 'Jurassic Park',
  Director: 'Steven Spielberg',
  Actors: 'Sam Neill, Laura Dern, Jeff Goldblum',
  Released: '11 Jun 1993',
  Poster: 'https://m.media-amazon.com/images/M/MV5BMjM2MDgxMDg0Nl5BMl5BanBnXkFtZTgwNTM2OTM5NDE@._V1_SX300.jpg',
  Genre: 'Action, Adventure, Sci-Fi',
  Plot: "A pragmatic paleontologist touring an almost complete theme park on an island in Central America is tasked with protecting a couple of kids after a power failure causes the park's cloned dinosaurs to run loose.",
},
{
  Title:'The Godfather',
  Director: 'Francis Ford Coppola',
  Actors: 'Marlon Brando, Al Pacino, James Caan',
  Released: '24 Mar 1972',
  Poster: 'https://m.media-amazon.com/images/M/MV5BM2MyNjYxNmUtYTAwNi00MTYxLWJmNWYtYzZlODY3ZTk3OTFlXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg',
  Genre: 'Crime, Drama',
  Plot: 'Don Vito Corleone, head of a mafia family, decides to hand over his empire to his youngest son, Michael. However, his decision unintentionally puts the lives of his loved ones in grave danger.',
},
{
  Title: 'Alien',
  Director: 'Ridley Scott',
  Actors: 'Sigourney Weaver, Tom Skerritt, John Hurt',
  Released: '22 Jun 1979',
  Poster: 'https://m.media-amazon.com/images/M/MV5BOGQzZTBjMjQtOTVmMS00NGE5LWEyYmMtOGQ1ZGZjNmRkYjFhXkEyXkFqcGdeQXVyMjUzOTY1NTc@._V1_SX300.jpg',
  Genre: 'Horror, Sci-Fi',
  Plot: 'The crew of a commercial spacecraft encounters a deadly lifeform after investigating a mysterious transmission of unknown origin.',
}];

describe('FilmsToWatchList', () => {

  it('should match snapshot', () => {
    render(<FilmsToWatchList filmsToWatch={films} removeFromStorage={null}/>);

    const filmContainer = document.querySelector('.filmCard') as HTMLElement;

    if (filmContainer) {
      expect(pretty(filmContainer.innerHTML)).toMatchSnapshot();
    }
  })
  
});
