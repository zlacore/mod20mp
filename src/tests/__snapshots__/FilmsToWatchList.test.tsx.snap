// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilmsToWatchList > should match snapshot 1`] = `
"<figure><img alt="Jurassic Park" src="https://m.media-amazon.com/images/M/MV5BMjM2MDgxMDg0Nl5BMl5BanBnXkFtZTgwNTM2OTM5NDE@._V1_SX300.jpg"></figure>
<article class="details">
  <h2>Jurassic Park</h2>
  <p><strong>Directed By:</strong> Steven Spielberg</p>
  <p><strong>Starring:</strong> Sam Neill, Laura Dern, Jeff Goldblum</p>
  <p><strong>Released:</strong> 11 Jun 1993</p>
  <p><strong>Genre:</strong> Action, Adventure, Sci-Fi</p>
</article>
<article class="plot">
  <p><strong>Plot:</strong> A pragmatic paleontologist touring an almost complete theme park on an island in Central America is tasked with protecting a couple of kids after a power failure causes the park's cloned dinosaurs to run loose.</p>
</article>
<aside class="icons"><svg stroke="currentColor" fill="currentColor" stroke-width="0" version="1.1" viewBox="0 0 16 16" style="font-size: 40px; cursor: pointer;" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg">
    <path d="M15.854 12.854c-0-0-0-0-0-0l-4.854-4.854 4.854-4.854c0-0 0-0 0-0 0.052-0.052 0.090-0.113 0.114-0.178 0.066-0.178 0.028-0.386-0.114-0.529l-2.293-2.293c-0.143-0.143-0.351-0.181-0.529-0.114-0.065 0.024-0.126 0.062-0.178 0.114 0 0-0 0-0 0l-4.854 4.854-4.854-4.854c-0-0-0-0-0-0-0.052-0.052-0.113-0.090-0.178-0.114-0.178-0.066-0.386-0.029-0.529 0.114l-2.293 2.293c-0.143 0.143-0.181 0.351-0.114 0.529 0.024 0.065 0.062 0.126 0.114 0.178 0 0 0 0 0 0l4.854 4.854-4.854 4.854c-0 0-0 0-0 0-0.052 0.052-0.090 0.113-0.114 0.178-0.066 0.178-0.029 0.386 0.114 0.529l2.293 2.293c0.143 0.143 0.351 0.181 0.529 0.114 0.065-0.024 0.126-0.062 0.178-0.114 0-0 0-0 0-0l4.854-4.854 4.854 4.854c0 0 0 0 0 0 0.052 0.052 0.113 0.090 0.178 0.114 0.178 0.066 0.386 0.029 0.529-0.114l2.293-2.293c0.143-0.143 0.181-0.351 0.114-0.529-0.024-0.065-0.062-0.126-0.114-0.178z"></path>
  </svg></aside>"
`;
